compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1368,6 +1368,17 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                         .source_map()
                         .span_take_while(span, |c| c.is_whitespace() || *c == '&');
                     if points_at_arg && mutability.is_not() && refs_number > 0 {
+                        // If we have a call like foo(&mut buf), then don't suggest foo(&mut mut buf)
+                        if "mut"
+                            == snippet
+                                .chars()
+                                .filter(|c| !c.is_whitespace())
+                                .skip(refs_number)
+                                .take(3)
+                                .collect::<String>()
```

Comment:
I really don't like the string comparison here, but if you're going to do it, then you can simplify it with something like `trim_start_matches` and `start_with` instead of collecting chars into a string.

Reply:
Yes, that considerably cleans it up.